core/src/main/java/org/elasticsearch/monitor/fs/FsInfo.java

```diff
@@ -496,10 +496,11 @@ public DiskUsage getMostDiskEstimate() {
     private Path total() {
         Path res = new Path();
         Set<String> seenDevices = new HashSet<>(paths.length);
+        Set<String> seenMounts = new HashSet<>(paths.length);
         for (Path subPath : paths) {
             if (subPath.path != null) {
-                if (!seenDevices.add(subPath.path)) {
-                    continue; // already added numbers for this device;
+                if (!seenDevices.add(subPath.path) || !seenMounts.add(subPath.mount)) {
```

Comment:
This doesn't look right to me. If `subPath.path` is null, or if `seenDevices.add()` returned false then we don't call `seenMounts.add()` despite having seen the mount point in question.

Reply:
Good point, let me work on it

Reply:
Based on your comments, I made the following assumptions:

1. valid path + valid mount point - OK
2. invalid path + valid mount point - OK
3. valid path + invalid mount point - NOT OK
4. invalid path + invalid mount point - NOT OK

Do update me if you think the above is incorrect.